### PR TITLE
Add an example of lossy decoding to str::Utf8Error docs

### PR DESCRIPTION
**Pull request by SimonSapin**
CC https://github.com/rust-lang/rust/issues/33906

### CODE REVIEW

File: src/libcore/str/mod.rs [hunk 1]
```diff
@@ -165,6 +165,40 @@ Section: Creating a string
...
+///                 break
+///             }
+///             Err(error) => {
+///                 let (valid, after_valid) = input.split_at(error.valid_up_to());
+///                 unsafe {
+///                     push(::std::str::from_utf8_unchecked(valid))
+///                 }
+///                 push("\u{FFFD}");
+///
+///                 match error.error_len() {
```
Comment (kennytm) [inline src/libcore/str/mod.rs]: Just stylistic comment. I think that this `match` taking 8 lines is artificially too long, you could make it 5 lines with `if let`.

```rust
if let Some(invalid_sequence_length) = error.error_len() {
    input = &after_valid[invalid_sequence_length..];
} else {
    break;
}
```

Comment (SimonSapin) [reply src/libcore/str/mod.rs]: This does look better. Applied, thanks.

**Pull request merged**